src/sponge.rs

```diff
@@ -6,36 +6,31 @@
 
 use crate::hash as zelbet;
 use crate::hash_gadget as zelbet_gadget;
-use crate::utils::ZelbetWitnesses;
-use dusk_plonk::{
-    constraint_system::{Constraint, TurboComposer, Witness},
-    prelude::BlsScalar,
-};
+use dusk_plonk::prelude::{BlsScalar, Constraint, TurboComposer, Witness};
 
 /// Sponge design for in circuit reinforced concrete, with arbitrary length
 /// input and output. Input length is read by the function, and output length is
 /// an input parameter.
 pub fn sponge_gadget(
     composer: &mut TurboComposer,
-    input: Vec<Witness>,
+    mut input: Vec<Witness>,
```

Comment:
We have no guarantee that the output size should be the same as the input. In fact, that is unlikely.
```suggestion
    input: &[Witness],
```

Reply:
The mutability is for the padding function. But I see that `sponge` needs closer attention. For now I reversed the changes in mutability of `input` (as it actually also not part of this PR anyway) and opened an issue for the changes in `sponge` #50